packages/esm-appointments-app/src/appointment-forms/appointments-forms.test.tsx

```diff
@@ -0,0 +1,116 @@
+import { mockPatient, mockServices, mockProviders } from '../../__mocks__/appointments.mock';
+import { mockLocations } from '../../../../__mocks__/locations.mock';
+import { mockSession } from '../../../../__mocks__/session.mock';
+import { MappedAppointment } from '../types';
+import AppointmentForm from './appointments-form.component';
+
+const mockedUseConfig = useConfig as jest.Mock;
+const mockedUsePatient = usePatient as jest.Mock;
+
+function renderAppointmentsForm(context: string, patientUuid?: string, appointment?: MappedAppointment) {
+  render(<AppointmentForm patientUuid={patientUuid} context={context} appointment={appointment} />);
+}
+
+jest.mock('./appointment-forms.resource.ts', () => {
+  const originalModule = jest.requireActual('./appointment-forms.resource.ts');
+
+  return {
+    ...originalModule,
+    useServices: jest.fn().mockReturnValue({ services: mockServices }),
+  };
+});
+
+jest.mock('@openmrs/esm-framework', () => {
+  const originalModule = jest.requireActual('@openmrs/esm-framework');
+  return {
+    ...originalModule,
+    openmrsFetch: jest.fn(),
+    useLocations: jest.fn().mockImplementation(() => mockLocations.data),
+    useSession: jest.fn().mockImplementation(() => mockSession.data),
+    useProviders: jest.fn().mockImplementation(() => mockProviders.data),
+  };
+});
+
+let mockOpenmrsConfig = {
+  daysOfTheWeek: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+  appointmentKinds: ['Scheduled', 'WalkIn', 'Virtual'],
+  appointmentStatuses: ['Requested', 'Scheduled', 'CheckedIn', 'Completed', 'Cancelled', 'Missed'],
+};
+
+describe('AppointmentForm', () => {
+  const patient = mockPatient;
+  beforeEach(() => {
+    mockedUseConfig.mockReturnValue(mockOpenmrsConfig);
+    mockedUsePatient.mockReturnValue({
+      patient,
+      isLoading: false,
+      error: null,
+      patientUuid: patient.uuid,
+    });
+  });
+
+  it('renders the patient banner with the right patient', () => {
+    renderAppointmentsForm('creating', mockPatient.uuid);
+
+    expect(screen.getByText(/Appointments Date and Time/i)).toBeInTheDocument();
+    // Need to test the banner extension
+  });
+
+  it('renders the form with all expected inputs in create mode', () => {
+    renderAppointmentsForm('creating', mockPatient.uuid);
+
+    const allInputs = screen.queryAllByLabelText(
+      (content, element) => element.tagName.toLowerCase() === 'input',
+    ) as Array<HTMLInputElement>;
+    const allSelects = screen.queryAllByRole('combobox') as Array<HTMLInputElement>;
+    let inputAndSelectNames = [];
+
+    allInputs.forEach((input) => inputAndSelectNames.push(input.id));
+    allSelects.forEach((select) => inputAndSelectNames.push(select.id));
+
+    expect(screen.queryByLabelText('Reason For Changes')).not.toBeInTheDocument();
+
+    expect(inputAndSelectNames).toEqual([
+      'visitStartDateInput',
+      'start-time-picker',
+      'end-time-picker',
+      'Yes',
+      'No',
+      '',
```

Comment:
There is an input missing an id?

Reply:
I can't seem to find it in the code but it expects it. I still need to write other tests for this form, so I will clean that up later.